model_check.py

```diff
@@ -0,0 +1,758 @@
+import os
+from pprint import pformat
+import re
+import time
+import pickle
+import numpy as np
+import Ska.DBI
+import Ska.Table
+import Ska.Numpy
+from Chandra.Time import DateTime
+import Chandra.cmd_states as cmd_states
+import matplotlib.pyplot as plt
+from Ska.Matplotlib import cxctime2plotdate, pointpair, plot_cxctime
+import Ska.engarchive.fetch_sci as fetch
+import Ska.Sun
+import shutil
+import glob
+import logging
+
+TASK_DATA = os.path.dirname(__file__)
+
+def calc_off_nom_rolls(states):
+    off_nom_rolls = []
+    for i, state in enumerate(states):
+        att = [state[x] for x in ['q1', 'q2', 'q3', 'q4']]
+        time = (state['tstart'] + state['tstop']) / 2
+        off_nom_rolls.append(Ska.Sun.off_nominal_roll(att, time))
+    return np.array(off_nom_rolls)
+
+class ModelCheck(object):
+    def __init__(self, msid, short_msid, MSIDs, yellow, margin,
+                 validation_limits, hist_limit, calc_model,
+                 version, other_telem=None, other_map=None,
+                 other_opts=None):
+        self.msid = msid
+        self.short_msid = short_msid
+        self.MSIDs = MSIDs
+        self.yellow = yellow
+        self.margin = margin
+        self.validation_limits = validation_limits
+        self.hist_limit = hist_limit
+        self.calc_model = calc_model
+        self.logger = logging.getLogger('%s_check' % self.short_msid)
+        self.version = version
+        self.other_telem = other_telem
+        self.other_map = other_map
+        self.other_opts = other_opts
+
+    def driver(self, opt):
+        if not os.path.exists(opt.outdir):
+            os.mkdir(opt.outdir)
+
+        config_logging(opt.outdir, opt.verbose, self.short_msid)
+
+        # Store info relevant to processing for use in outputs
+        proc = dict(run_user=os.environ['USER'],
+                    run_time=time.ctime(),
+                    errors=[],
+                    msid=self.msid.upper(),
+                    short_msid=self.short_msid.upper(),
+                    hist_limit=self.hist_limit)
+        proc["msid_limit"] = self.yellow[self.short_msid] - self.margin[self.short_msid]
+        self.logger.info('##############################'
+                    '#######################################')
+        self.logger.info('# %s_check.py run at %s by %s'
+                    % (self.short_msid, proc['run_time'], proc['run_user']))
+        self.logger.info('# %s_check version = %s' % (self.short_msid, self.version))
+        self.logger.info('# model_spec file = %s' % os.path.abspath(opt.model_spec))
+        self.logger.info('###############################'
+                    '######################################\n')
+
+        self.logger.info('Command line options:\n%s\n' % pformat(opt.__dict__))
+
+        # Connect to database (NEED TO USE aca_read)
+        self.logger.info('Connecting to database to get cmd_states')
+        db = Ska.DBI.DBI(dbi='sybase', server='sybase', user='aca_read',
```

Comment:
There's the plan to wean these checkers off the 'sybase' server.  Should that be a priority for a first release of the new model_check or a later enhancement?